frontend/src/lib/k8s/api/v2/useKubeObjectList.ts

```diff
@@ -348,12 +348,13 @@ export function makeListRequests(
   clusters: string[],
   getAllowedNamespaces: (cluster: string | null) => string[],
   isResourceNamespaced: boolean,
-  requestedNamespaces?: string[]
+  requestedNamespaces: string[] = []
 ): Array<{ cluster: string; namespaces?: string[] }> {
   return clusters.map(cluster => {
     const allowedNamespaces = getAllowedNamespaces(cluster);
 
-    let namespaces = requestedNamespaces ?? allowedNamespaces;
+    let namespaces = requestedNamespaces.length > 0 ? requestedNamespaces : allowedNamespaces;
```

Comment:
Should we double check if the requested namespaces are contained in the allowedNamespaces, when both are set?

Reply:
namespaces are filtered below to only contain allowedNamespaces (if they are set)